api/v1alpha1/clientprofilemapping_types.go

```diff
@@ -20,29 +20,22 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// BlockPoolRefSpec identify a blockpool - client profile pair
-type BlockPoolRefSpec struct {
-	//+kubebuilder:validation:Required
-	ClientProfileName string `json:"clientProfileName,omitempty"`
-
-	//+kubebuilder:validation:Required
-	//+kubebuilder:validation:Minimum:=0
-	PoolId int `json:"poolId,omitempty"`
-}
-
-// BlockPoolMappingSpec define a mapiing between a local and remote block pools
-type BlockPoolMappingSpec struct {
-	//+kubebuilder:validation:Required
-	Local BlockPoolRefSpec `json:"local,omitempty"`
-
-	//+kubebuilder:validation:Required
-	Remote BlockPoolRefSpec `json:"remote,omitempty"`
+// MappinsSpec define a mapiing between a local and remote profiles 
+type MappingsSepc struct {
+    //+kubebuilder:validation:Required
+    LocalClientProfile string `json:"localClientProfile,omitempty"`
+    
+    //+kubebuilder:validation:Required
+    RemoteClientProfile string `json:"remoteClientProfile,omitempty"`
+
+    //+kubebuilder:validation:Optional
+    BlockPoolIdMapping map[string]string `json:"blockPoolIdMapping,omitempty"`
```

Comment:
List of `BlockPoolIdMapping`?  @Madhu-1

Reply:
Why do we need a list here? I know the serialized form has a list of maps, but it is not being used in any/all examples or use cases known to be.  If there is a use case that cannot be captured with a single map, please raise it up in a reply

Reply:
if we have a map we cannot have duplicate keys lets say the poolID `1` needs mapping to both poolID `8` and poolID `9`

Reply:
Ok, in that case we don't need a map at all, I will change to an array of pairs
@Madhu-1 @sp98 agreed?

Reply:
sounds good 👍🏻